### PR TITLE
`SiPixelLorentzAnglePCLHarvester`: fetch precise B-field value from map via `inverseBzAtOriginInGeV`

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

The primary goal of this PR is to address the second point of issue https://github.com/cms-sw/cmssw/issues/38669:

> Unrelated to the crash, but pixel people beware: using magField->nominalValue() as a physical value as in https://github.com/cms-sw/cmssw/pull/38653/files#diff-cd4b91b1150cb8ea3ac6cdb24d49a6bd99f77a8320a5d4a60eeab3000ff42eb1R141 is not a good idea. `nominalValue()` is intended to be used as a label, and is truncated to int, with the corresponding loss of accuracy.
The method `inverseBzAtOriginInGeV()` was added to the interface for the benefit of pixel code (https://github.com/cms-sw/cmssw/pull/35081), for example.

This is done by taking the value of the magnetic field from `inverseBzAtOriginInGeV()` (after having converted it from [GeV<sup>-1</sup>]  to [T]) instead of `nominalValue()`.
I profit of this PR to rename all the other member variable to follow the code rule pattern (rule 2.9: https://cms-sw.github.io/cms_coding_rules.html)

#### PR validation:

Tested successfully the multi-run harvesting mode recipe at https://github.com/cms-sw/cmssw/pull/38641#issuecomment-1178053372;

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Not a backport, but a backport to CMSSW_12_4_X is needed.

### CODE REVIEW

**Comment by tvami** [comment]
There is a few lines of commented out code, that should be removed, but besides that it looks good to me, thanks @mmusich !

**Comment by mmusich** [comment]
> There is a few lines of commented out code, that should be removed, 

why? they are you know.. a comment...

**Comment by tvami** [comment]
> why? they are you know.. a comment...

Ok, ignore my comment, I was reading through the code too quickly

**Comment by tvami** [comment]
type bugfix,trk

**Comment by mmusich** [comment]
please test

**Comment by tvami** [comment]
+alca
 * PR is according to the description 
 * Jenkins test pass

File: CalibTracker/SiPixelLorentzAngle/src/SiPixelLorentzAnglePCLHarvester.cc [hunk 1]
```diff
@@ -100,13 +100,15 @@ class SiPixelLorentzAnglePCLHarvester : public DQMEDHarvester {
...
   const std::string recordName_;
-  std::unique_ptr<TF1> f1;
+  std::unique_ptr<TF1> f1_;
   float width_;
   float theMagField_{0.f};
 
-  SiPixelLorentzAngleCalibrationHistograms hists;
-  const SiPixelLorentzAngle* currentLorentzAngle;
-  std::unique_ptr<TrackerTopology> theTrackerTopology;
+  static constexpr float inverseGeVtoTesla_ = 2.99792458e-3f;
```
Comment (ferencek) [inline CalibTracker/SiPixelLorentzAngle/src/SiPixelLorentzAnglePCLHarvester.cc:107]: Wouldn't `inverseGeVperTesla` or `teslaToInverseGeV` be a more appropriate name for this conversion factor? (sorry for a late comment, catching up with stuff that was happening while being away on vacation)

Comment (mmusich) [reply CalibTracker/SiPixelLorentzAngle/src/SiPixelLorentzAnglePCLHarvester.cc:107]: Won't bother now. Seems ok to me.

**Pull request merged**